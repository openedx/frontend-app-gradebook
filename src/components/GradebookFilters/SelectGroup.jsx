/* eslint-disable react/sort-comp, react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';

import { Form } from '@edx/paragon';

const SelectGroup = ({
  id,
  label,
  value,
  onChange,
  disabled,
  options,
}) => (
  <div className="student-filters">
    <Form.Group controlId={id}>
      <Form.Label>{label}</Form.Label>
      <Form.Control as="select" {...{ value, onChange, disabled }}>
        {options}
      </Form.Control>
    </Form.Group>
  </div>
);
SelectGroup.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.node.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  options: PropTypes.arrayOf(PropTypes.node).isRequired,
};
SelectGroup.defaultProps = {
  disabled: false,
};

export default SelectGroup;
