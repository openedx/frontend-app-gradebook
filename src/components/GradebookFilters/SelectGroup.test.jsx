import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import SelectGroup from './SelectGroup';

describe('SelectGroup', () => {
  let props = {
    id: 'group id',
    label: 'Group Label',
    value: 'group VALUE',
    disabled: false,
    options: [
      <option value="opt1" key="opt1">Option 1</option>,
      <option value="opt2" key="opt2">Option 2</option>,
      <option value="opt3" key="opt3">Option 3</option>,
    ],
  };

  beforeEach(() => {
    props = {
      ...props,
      onChange: jest.fn().mockName('props.onChange'),
    };
  });

  describe('Component', () => {
    describe('snapshots', () => {
      test('basic snapshot', () => {
        const el = shallow(<SelectGroup {...props} />);
        expect(el.snapshot).toMatchSnapshot();
      });
      test('disabled', () => {
        const el = shallow(<SelectGroup {...props} disabled />);
        expect(el.snapshot).toMatchSnapshot();
      });
    });
  });
});
