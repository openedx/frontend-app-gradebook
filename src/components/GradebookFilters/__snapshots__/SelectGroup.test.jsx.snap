// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectGroup Component snapshots basic snapshot 1`] = `
<div
  className="student-filters"
>
  <Form.Group
    controlId="group id"
  >
    <Form.Label>
      Group Label
    </Form.Label>
    <Form.Control
      as="select"
      disabled={false}
      onChange={[MockFunction props.onChange]}
      value="group VALUE"
    >
      <option
        key="opt1"
        value="opt1"
      >
        Option 1
      </option>
      <option
        key="opt2"
        value="opt2"
      >
        Option 2
      </option>
      <option
        key="opt3"
        value="opt3"
      >
        Option 3
      </option>
    </Form.Control>
  </Form.Group>
</div>
`;

exports[`SelectGroup Component snapshots disabled 1`] = `
<div
  className="student-filters"
>
  <Form.Group
    controlId="group id"
  >
    <Form.Label>
      Group Label
    </Form.Label>
    <Form.Control
      as="select"
      disabled={true}
      onChange={[MockFunction props.onChange]}
      value="group VALUE"
    >
      <option
        key="opt1"
        value="opt1"
      >
        Option 1
      </option>
      <option
        key="opt2"
        value="opt2"
      >
        Option 2
      </option>
      <option
        key="opt3"
        value="opt3"
      >
        Option 3
      </option>
    </Form.Control>
  </Form.Group>
</div>
`;
